cli/cmd/vhost.go

```diff
@@ -0,0 +1,178 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"regexp"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/OJ/gobuster/v3/cli"
+	"github.com/OJ/gobuster/v3/gobustervhost"
+	"github.com/OJ/gobuster/v3/libgobuster"
+	"github.com/spf13/cobra"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+var cmdVhost *cobra.Command
+
+func runVhost(cmd *cobra.Command, args []string) error {
+	globalopts, pluginopts, err := parseVhostOptions()
+	if err != nil {
+		return fmt.Errorf("error on parsing arguments: %v", err)
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	plugin, err := gobustervhost.NewGobusterVhost(ctx, globalopts, pluginopts)
+	if err != nil {
+		return fmt.Errorf("error on creating gobustervhost: %v", err)
+	}
+
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, os.Interrupt)
+	go func() {
+		for range signalChan {
+			// caught CTRL+C
+			if !globalopts.Quiet {
+				fmt.Println("\n[!] Keyboard interrupt detected, terminating.")
+			}
+			cancel()
+		}
+	}()
+
+	if err := cli.Gobuster(ctx, globalopts, plugin); err != nil {
+		return fmt.Errorf("error on running goubster: %v", err)
+	}
+	return nil
+}
+
+func parseVhostOptions() (*libgobuster.Options, *gobustervhost.OptionsVhost, error) {
+	globalopts, err := parseGlobalOptions()
+	if err != nil {
+		return nil, nil, err
+	}
+	var plugin gobustervhost.OptionsVhost
+
+	url, err := cmdVhost.Flags().GetString("url")
+	if err != nil {
+		return nil, nil, fmt.Errorf("invalid value for url: %v", err)
+	}
+	plugin.URL = url
+	if !strings.HasPrefix(plugin.URL, "http") {
+		// check to see if a port was specified
+		re := regexp.MustCompile(`^[^/]+:(\d+)`)
+		match := re.FindStringSubmatch(plugin.URL)
+
+		if len(match) < 2 {
+			// no port, default to http on 80
+			plugin.URL = fmt.Sprintf("http://%s", plugin.URL)
+		} else {
+			port, err2 := strconv.Atoi(match[1])
+			if err2 != nil || (port != 80 && port != 443) {
+				return nil, nil, fmt.Errorf("url scheme not specified")
+			} else if port == 80 {
+				plugin.URL = fmt.Sprintf("http://%s", plugin.URL)
+			} else {
+				plugin.URL = fmt.Sprintf("https://%s", plugin.URL)
+			}
+		}
+	}
+
+	cookies, err := cmdVhost.Flags().GetString("cookies")
```

Comment:
Probably no need to have temp variables every time we parse a flag. I think it makes more sense to write them straight to the plugin content instead. We throw errors if they're invalid anyway.
```go
plugin.Cookies, err = cmdVhost.Flags().GetString("cookies")
```
Same goes for all the options parsed below.